akd_local_auditor/src/storage/s3/mod.rs

```diff
@@ -45,6 +51,11 @@ fn is_bucket_name_valid(s: &str) -> Result<String, String> {
     Ok(str)
 }
 
+fn is_uri_valid(s: &str) -> Result<String, String> {
+    let _uri: http::Uri = s.parse::<http::Uri>().map_err(|err| err.to_string())?;
+    Ok(s.to_string())
+}
```

Comment:
This feels like sort of an odd name given we're not returning a `bool`, but instead validating a URI and returning a Result. Perhaps something more reflective of what's happening here would be `validate_uri` or `parse_uri`.

Reply:
Yes, clap requires value parsers to return `Result<String, String>` not a bool type. But because of that we can definitely use a better name